datalad/support/annexrepo.py

```diff
@@ -0,0 +1,61 @@
+"""
+Interface to git-annex by Joey Hess.
+See https://git-annex.branchable.com/.
+"""
+__author__ = 'Benjamin Poldrack'
+
+import os
+from os.path import join, exists
+
+from gitrepo import GitRepo
+import datalad.log
+
+from datalad.cmd import Runner
+
+
+class AnnexRepo(GitRepo):
+    """
+    Representation of an git-annex repository.
+
+    """
+
+    def __init__(self, path, url=None):
+        """
+        AnnexRepo is initialized by giving a path to the annex.
+        If no annex exists at that location, a new one is created.
+        Optionally give url to clone from.
+
+        :param path: path to git-annex repository
+        :param url: valid git url. See http://www.kernel.org/pub/software/scm/git/docs/git-clone.html#URLS
+        :return:
+        """
+        super(AnnexRepo, self).__init__(path, url)
+
+        self.cmd_call_wrapper = Runner()
+        # TODO: Concept of when to set to "dry". Includes: What to do in gitrepo class?
+
+
+        # Check whether an annex already exists at destination
+        if not exists(join(self.path, '.git', 'annex')):
+            datalad.log.lgr.debug('No annex found in %s. Creating a new one ...' % self.path)
+            self.annex_init()
+
+    def annex_init(self):
+        """
+        Invokes command 'git annex init'.
+
+        :return:
+        """
+
+        status, output = self.cmd_call_wrapper.getstatusoutput('cd %s && git annex init' % self.path)
```

Comment:
What about windows then? Do they have cd now?
On Feb 24, 2015 6:56 AM, "Yaroslav Halchenko" <notifications@github.com>
wrote:

> In datalad/support/annexrepo.py
> <https://github.com/datalad/datalad/pull/42#discussion_r25230534>:
>
> > +        # TODO: Concept of when to set to "dry". Includes: What to do in gitrepo class?
> > +
> > +
> > +        # Check whether an annex already exists at destination
> > +        if not exists(join(self.path, '.git', 'annex')):
> > +            datalad.log.lgr.debug('No annex found in %s. Creating a new one ...' % self.path)
> > +            self.annex_init()
> > +
> > +    def annex_init(self):
> > +        """
> > +        Invokes command 'git annex init'.
> > +
> > +        :return:
> > +        """
> > +
> > +        status, output = self.cmd_call_wrapper.getstatusoutput('cd %s && git annex init' % self.path)
>
>  > + status, output = self.cmd_call_wrapper.getstatusoutput('cd %s && git
> annex init' % self.path) why 'cd? and not os.chdir?
> because that switches entire process to that directory. What for? you
> would need to the try/except and cd back (remember curdir before, possible
> problems with symlinked directories etc) -- so is it worth it? ;) Better
> have that shell process switch where necessary
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/42/files#r25230534>.
>